From da3273a1abe44eb324b609a88d24fa29055d2cea Mon Sep 17 00:00:00 2001
From: Prashant Singh Chauhan <prashant.singh-chauhan@broadcom.com>
Date: Fri, 27 Mar 2026 07:52:17 +0000
Subject: [PATCH] make default openssl legacy false

If CRYPTOGRAPHY_OPENSSL_NO_LEGACY is not set or set to 1, we would not load openssl
legacy crypto.
If CRYPTOGRAPHY_OPENSSL_NO_LEGACY is set to 0, we will load openssl legacy crypto

---
 src/rust/src/lib.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rust/src/lib.rs b/src/rust/src/lib.rs
index 97f4dc11a..e05c3b7f3 100644
--- a/src/rust/src/lib.rs
+++ b/src/rust/src/lib.rs
@@ -63,7 +63,7 @@ fn _initialize_providers(py: pyo3::Python<'_>) -> CryptographyResult<LoadedProvi
     // any time soon.
 
     let load_legacy = !cfg!(CRYPTOGRAPHY_BUILD_OPENSSL_NO_LEGACY)
-        && !env::var("CRYPTOGRAPHY_OPENSSL_NO_LEGACY").is_ok_and(|v| !v.is_empty() && v != "0");
+        && env::var("CRYPTOGRAPHY_OPENSSL_NO_LEGACY").is_ok_and(|v| v == "0");
 
     let legacy = if load_legacy {
         let legacy_result = provider::Provider::load(None, "legacy");
-- 
2.43.7

